packages/flutter/lib/src/widgets/spell_check.dart

```diff
@@ -201,20 +211,106 @@ TextSpan buildTextSpanWithSpellCheckSuggestions(
         value.text, spellCheckResultsText, spellCheckResultsSpans);
   }
 
-  return TextSpan(
+  // We will draw the TextSpan tree based on the composing region, if it is
+  // available.
+  final bool shouldConsiderComposingRegion = defaultTargetPlatform == TargetPlatform.android;
+  if (shouldConsiderComposingRegion) {
+    return TextSpan(
       style: style,
-      children: _buildSubtreesWithMisspelledWordsIndicated(
+      children: _buildSubtreesWithComposingRegion(
           spellCheckResultsSpans,
           value,
           style,
           misspelledTextStyle,
-          composingWithinCurrentTextRange
+          composingWithinCurrentTextRange,
+      ),
+    );
+  }
+
+  return TextSpan(
+    style: style,
+    children: _buildSubtreesWithoutComposingRegion(
```

Comment:
I just tried it: It works the same, but the composing region doesn't draw. I think that's expected behavior!

Reply:
That makes me think that we could use only this method and get rid of the other one.  It could remove the composing region when it's part of the replaced word and then draw it.  Maybe in another PR if that's going to delay this one, though.

Reply:
I definitely think that they can be merged into one and is worth pursuing, then. I am running short on time, though, so unless you're open to playing around with it, I say we delay it for now and file an issue!